packages/strapi-plugin-content-type-builder/admin/src/components/AllowedTypesSelect/MenuList.js

```diff
@@ -0,0 +1,129 @@
+    },
+  ];
+
+  return (
+    <Component {...rest}>
+      <Ul
+        style={{
+          maxHeight: 150,
+        }}
+      >
+        <li className="li-multi-menu">
+          <div style={{ marginTop: 3 }}>
+            <CheckboxWrapper>
+              <Label
+                htmlFor="overrideReactSelectBehavior"
+                onClick={() => {
+                  changeMediaAllowedTypes({
+                    target: { name: 'all', value: !areAllAllowedTypesSelected },
+                  });
+                }}
+              >
+                <Checkbox
+                  id="checkAll"
+                  checked={areAllAllowedTypesSelected}
+                  someChecked={someChecked}
+                  name="all"
+                  onChange={() => {}}
+                  style={{ marginRight: 10 }}
+                />
+                <UpperFirst content="All" />
+              </Label>
+            </CheckboxWrapper>
+          </div>
+          <SubUl tad="ul" isOpen>
+            {options.map(({ name, infos }) => {
+              const isChecked = value.value.includes(name);
+              const target = { name, value: !isChecked };
+
+              return (
+                <li key={name}>
+                  <CheckboxWrapper>
+                    <Label
+                      htmlFor={name}
+                      onClick={() => {
+                        changeMediaAllowedTypes({ target });
+                      }}
+                    >
+                      <Checkbox
+                        id="check"
+                        name={name}
+                        // Remove the handler
+                        onChange={() => {}}
+                        checked={isChecked}
+                        style={{ marginRight: 10 }}
+                      />
+                      <span style={{ display: 'contents' }}>
+                        <UpperFirst
+                          content={formatMessage({
+                            id: getTrad(`form.attribute.media.allowed-types.option-${name}`),
+                          })}
+                        />
+                      </span>
+                      <span
+                        style={{
+                          display: 'contents',
+                          fontSize: 11,
+                          color: '#B3B5B9',
+                          textTransform: 'italic',
+                        }}
+                      >
+                        &nbsp;{infos}
+                      </span>
+                    </Label>
+                  </CheckboxWrapper>
+                </li>
+              );
+            })}
+          </SubUl>
+        </li>
+      </Ul>
+    </Component>
+  );
+};
+
+MenuList.defaultProps = {
+  selectProps: {
+    value: {},
+  },
+};
+
+MenuList.propTypes = {
+  selectProps: PropTypes.shape({
+    changeMediaAllowedTypes: PropTypes.func.isRequired,
+    value: PropTypes.object,
+  }),
+};
+
+export default MenuList;
```

Comment:
There is a lot of inline style customization here. We should avoid this for the app performance and for the scalability.

Reply:
I have created new components